crates/sui-types/src/message_envelope.rs

```diff
@@ -177,62 +171,59 @@ where
     pub fn epoch(&self) -> EpochId {
         self.auth_signature.epoch
     }
+}
 
-    pub fn verify_committee_sigs_only(&self, committee: &Committee) -> SuiResult
-    where
-        <T as Message>::DigestType: PartialEq,
-    {
-        self.data.verify_epoch(self.auth_sig().epoch)?;
-        self.auth_signature
-            .verify_secure(self.data(), Intent::sui_app(T::SCOPE), committee)
+impl Envelope<SenderSignedData, AuthoritySignInfo> {
+    pub fn verify_committee_sigs_only(&self, committee: &Committee) -> SuiResult {
+        self.auth_signature.verify_secure(
+            self.data(),
+            Intent::sui_app(IntentScope::SenderSignedTransaction),
+            committee,
+        )
     }
 }
 
-impl<T> Envelope<T, AuthoritySignInfo>
-where
-    T: Message + AuthenticatedMessage + Serialize,
-{
+impl SignedTransaction {
     pub fn verify_signatures_authenticated(
         &self,
         committee: &Committee,
         verify_params: &VerifyParams,
     ) -> SuiResult {
-        self.data.verify_epoch(self.auth_sig().epoch)?;
-        self.data.verify_message_signature(verify_params)?;
-        self.auth_signature
-            .verify_secure(self.data(), Intent::sui_app(T::SCOPE), committee)
+        verify_sender_signed_data_message_signatures(&self.data, committee.epoch(), verify_params)?;
+
+        self.auth_signature.verify_secure(
+            self.data(),
+            Intent::sui_app(IntentScope::SenderSignedTransaction),
+            committee,
+        )
     }
 
     pub fn verify_authenticated(
         self,
         committee: &Committee,
         verify_params: &VerifyParams,
-    ) -> SuiResult<VerifiedEnvelope<T, AuthoritySignInfo>> {
+    ) -> SuiResult<VerifiedSignedTransaction> {
         self.verify_signatures_authenticated(committee, verify_params)?;
-        Ok(VerifiedEnvelope::<T, AuthoritySignInfo>::new_from_verified(
-            self,
-        ))
+        Ok(VerifiedSignedTransaction::new_from_verified(self))
     }
 }
 
-impl<T> Envelope<T, AuthoritySignInfo>
-where
-    T: Message + UnauthenticatedMessage + Serialize,
-{
+impl SignedCheckpointSummary {
```

Comment:
This should be in messages_checkpoint.rs

Reply:
yeah, meant to move these, thanks